src/components/StakingApplicationForms/index.tsx

```diff
@@ -0,0 +1,175 @@
+  ...formikProps
+}) => {
+  const { values } = formikProps
+  const { tokenAmount } = values
+  const [remainingAmount, setRemainingAmount] = useState("0")
+  const [maxAmount, setMaxAmount] = useState("0")
+
+  useEffect(() => {
+    if (!isAuthorization) {
+      setMaxAmount(authorizedAmount)
+    } else {
+      setMaxAmount(BigNumber.from(totalStake).sub(authorizedAmount).toString())
+    }
+  }, [authorizedAmount, totalStake, isAuthorization])
+
+  useEffect(() => {
+    const _tokenAmount = BigNumber.from(tokenAmount || "0")
+    const _max = BigNumber.from(maxAmount)
+    setRemainingAmount(
+      tokenAmount
+        ? _tokenAmount.gte(_max)
+          ? "0"
+          : _max.sub(_tokenAmount).toString()
+        : maxAmount
+    )
+  }, [tokenAmount, maxAmount])
+
+  return (
+    <TokenAmountFormBase
+      label={
+        <Label
+          label={isAuthorization ? "Increase Amount" : "Decrease Amount"}
+          remainingAmount={remainingAmount}
+        />
+      }
+      submitButtonText={submitButtonText}
+      isDisabled={isDisabled}
+      maxTokenAmount={maxAmount}
+      placeholder={"Enter amount"}
+      helperText={helperText}
+      {...formikProps}
+    />
+  )
+}
+
+type StakingAppFormBaseProps = {
+  initialAmount?: string
+  minimumAuthorizationAmount: string
+  innerRef?: Ref<FormikProps<FormValues>>
+  onSubmitForm: (tokenAmount: string) => void
+} & ComponentProps
+
+const authorizationValidation = (
+  values: FormValues,
+  props: StakingAppFormBaseProps
+) => {
+  const errors: FormikErrors<FormValues> = {}
+
+  const { tokenAmount } = values
+  const { authorizedAmount, totalStake, minimumAuthorizationAmount } = props
+
+  const authorizedAmountInBN = BigNumber.from(authorizedAmount)
+  const max = BigNumber.from(totalStake).sub(authorizedAmountInBN)
+  const minimumAuthorizationAmountInBN = BigNumber.from(
+    minimumAuthorizationAmount
+  )
+
+  const min = authorizedAmountInBN.gt(minimumAuthorizationAmountInBN)
+    ? DEFAULT_MIN_VALUE
+    : minimumAuthorizationAmountInBN.sub(authorizedAmountInBN)
+
+  errors.tokenAmount = validateAmountInRange(
+    tokenAmount,
+    max.toString(),
+    min.toString()
+  )
+
+  return getErrorsObj(errors)
+}
+
+const deauthorizationValidation = (
+  values: FormValues,
+  props: StakingAppFormBaseProps
+) => {
+  const errors: FormikErrors<FormValues> = {}
+
+  const { tokenAmount } = values
+  const { authorizedAmount, minimumAuthorizationAmount } = props
+  const max = BigNumber.from(authorizedAmount).sub(minimumAuthorizationAmount)
+  const tokenAmountInBN = BigNumber.from(tokenAmount || "0")
+
+  if (!tokenAmountInBN.eq(authorizedAmount)) {
+    errors.tokenAmount = validateAmountInRange(
+      tokenAmount,
+      max.toString(),
+      DEFAULT_MIN_VALUE,
+      {
+        ...defaultValidationOptions,
+        lessThanValidationMsg(amount) {
+          return `${defaultLessThanMsg(amount)} or ${formatTokenAmount(
+            authorizedAmount.toString()
```

Comment:
I think this would be more readable:

```suggestion
        lessThanValidationMsg(amount) {
          return `${defaultLessThanMsg(amount)} or equal to ${formatTokenAmount(
            authorizedAmount.toString()
```

In your case it might indicate that the value should be less than or equal `amount` or less than or equal `authorized` amount.